pkgs/development/libraries/science/math/mkl/default.nix

```diff
@@ -51,18 +60,15 @@ stdenvNoCC.mkDerivation rec {
       cp license.txt $out/lib/
   '';
 
+  postFixup = stdenvNoCC.lib.optionalString stdenvNoCC.isDarwin ''
+      install_name_tool -change @rpath/libiomp5.dylib $out/lib/libiomp5.dylib $out/lib/libmkl_intel_thread.dylib
+      install_name_tool -change @rpath/libtbb.dylib ${tbb.out}/lib/libtbb.dylib $out/lib/libmkl_tbb_thread.dylib
+  '';
+
   # Per license agreement, do not modify the binary
   dontStrip = true;
   dontPatchELF = true;
 
-  # Since these are unmodified binaries from Intel, they do not depend on stdenv
-  # and we can make them fixed-output derivations for cache efficiency.
-  outputHashAlgo = "sha256";
-  outputHashMode = "recursive";
-  outputHash = if stdenvNoCC.isDarwin
-    then "00d49ls9vcjan1ngq2wx2q4p6lnm05zwh67hsmj7bnq43ykrfibw"
-    else "1amagcaan0hk3x9v7gg03gkw02n066v4kmjb32yyzsy5rfrivb1a";
-
```

Comment:
This package is huge (~600-700MB), and since it's just a straight RPM unpack on Linux with no other runtime dependencies, we were computing the output hash to avoid package rebuilds + re-downloads on changes to tools like `rpmextract`, `stdenvNoCC`, etc.  Since the Darwin package now depends on `tbb` we can't do that for Darwin anymore, but it seems reasonable to leave it in for Linux.  Perhaps we should keep these attributes for Linux via an optional dictionary merge?

Reply:
Fine with me, but I don't know how to implement that.

Reply:
Perhaps it should be possible to keep most of the things in a fixed-output derivation on macOS as well if we move rpmextract/undmg into the `postFetch` of the `fetchurl`. Then the derivation could symlink most of the files and patch necessary libraries.

Reply:
Ah that's smart @veprbl!  @smaret I'll try to sketch out some code and post it in the PR, but since I don't have access to a Mac it might be a little iffy.  With the `postFetch` trick I think we can clean this up a bit for both Linux and Mac.

Reply:
Perhaps for now we could go for a compromise and keep it fixed-hash at least on linux by doing something like:
```
stdenvNoCC.mkDerivation ({
  # ...
} // stdenvNoCC.lib.optionalAttrs stdenvNoCC.isLinux {
  outputHashAlgo = ...;
  # ...
})
```

Reply:
Done, thanks for the suggestion.